pkg/cloud/scope/machine.go

```diff
@@ -126,14 +128,15 @@ func (m *MachineScope) Region() string {
 	return m.Parent.Region()
 }
 
-// GetInstanceID returns the AWSMachine instance id from the status.
+// GetInstanceID returns the AWSMachine instance id by parsing Spec.ProviderID.
 func (m *MachineScope) GetInstanceID() *string {
-	return m.ProviderMachine.Status.InstanceID
-}
-
-// SetInstanceID sets the AWSMachine instance id.
-func (m *MachineScope) SetInstanceID(v string) {
-	m.ProviderMachine.Status.InstanceID = pointer.StringPtr(v)
+	if pid := m.GetProviderID(); pid != nil && *pid != "" {
+		// Get the ProviderID, and if it's populated, return the last part.
```

Comment:
Please use the ProviderID code from CAPI

Reply:
What do you mean?

Reply:
```go
parsed, err := noderefutil.NewProviderID(*pid)
if err != nil {
  // log and return nil?
}
return pointer.StringPtr(parsed.ID())
```

Reply:
Done and rewrote a little

Reply:
This still has the string checking?

Reply:
I'd avoid to use the noderefutil here if possible given that the GetInstanceID might be called frequently

Reply:
What is your concern?

Reply:
I guess it doesn't matter, but it seemed small enough that it wasn't worth to change